File: base/src/main/java/tbs/framework/utils/LogFactory.java
package tbs.framework.utils;

import tbs.framework.log.ILogger;

/**
 * @author abstergo
 */
public abstract class LogFactory {

    /**
     *
     */
    private static LogFactory logFactory = null;

    /**
     *
     */
    public static LogFactory getInstance() {
        return LogFactory.logFactory;
    }

    /**
     *
     */
    protected LogFactory() {
    }

    public static void setLogFactory(LogFactory logFactory) {
        LogFactory.logFactory = logFactory;
    }

    /**
     * 获取日志器
     *
     * @param name 日志器名，理论上应唯一
     * @return 日志器
     */
    public abstract ILogger getLogger(final String name);

}

File: baseParent/base/src/main/kotlin/tbs/framework/base/utils/LogFactory.kt
package tbs.framework.base.utils

import tbs.framework.log.ILogger

/**
 * @author abstergo
 */
abstract class LogFactory
/**
 *
 */
protected constructor() {
    /**
     * 获取日志器
     *
     * @param name 日志器名，理论上应唯一
     * @return 日志器
     */
    abstract fun getLogger(name: String? = null): ILogger

    companion object {
        /**
         *
         */
        @JvmStatic
        var instance: LogFactory? = null
            private set


        @JvmStatic
        fun setLogFactory(logFactory: LogFactory) {
            instance = logFactory
        }
    }
}
